AK/Optional.h

```diff
@@ -55,16 +59,37 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> {
 
     static_assert(!IsLvalueReference<T> && !IsRvalueReference<T>);
 
+    // FIXME: GCC seems to have an issue with uninitialized unions
+    //        which forces us to have an equally sized trivial null member in the union
+    //        to pseudo-initialize the union,
+    //        Note this could/should be an AK::Array, but it can't as that depends on Optional,
+    //        we use an anonymous struct with a dummy member instead
+    constexpr ALWAYS_INLINE void construct_null()
```

Comment:
extremely nitpicky nit you don't necessarily need to address:
```
❯ rg 'constexpr.*ALWAYS_INLINE' | wc -l
       3

❯ rg 'ALWAYS_INLINE.*constexpr' | wc -l
     128
```

how about `ALWAYS_INLINE constexpr`?